lib/src/data_table_parser.dart

```diff
@@ -1,37 +1,79 @@
+      if (isStep && isNextLineTable && !isExamplesFormatted) {
+        return true;
+      }
     }
-    if (i + 1 < lines.length && _foundDataTable(lines, i)) {
-      final dataTable = [
-        lines[i],
-        ...lines.skip(i + 1).takeWhile((l) => l.type == LineType.examples),
-      ];
-      final data = generateScenariosFromScenaioOutline([
-        // pretend to be an Example section to re-use some logic
-        BddLine.fromValue(LineType.exampleTitle, ''),
-        ...dataTable,
-      ]);
-      for (final item in data) {
-        yield item[1];
+    return false;
+  }
+
+  static Iterable<BddLine> replaceDataTables(List<BddLine> lines) sync* {
+    for (var index = 0; index < lines.length; index++) {
+      final isStep = lines[index].type == LineType.step;
+      final isNextLineTable = isNextTable(lines: lines, index: index + 1);
+      if (isStep && isNextLineTable) {
+        if (!hasExamplesFormat(bddLine: lines[index])) {
+          // Cucumber data table
+          final text = lines[index].value;
+          final table = List<List<String>>.empty(growable: true);
+          do {
+            table.add(_createRow(bddLine: lines[++index]));
+          } while (isNextTable(lines: lines, index: index + 1));
+          yield BddLine.fromValue(
+            LineType.step,
+            '$text {const bdd.DataTable($table)}',
+          );
+        } else {
+          // Data table formatted as examples
+          final dataTable = [lines[index]];
+          do {
+            dataTable.add(lines[++index]);
+          } while (index + 1 < lines.length &&
+              lines[index + 1].type == LineType.examples);
+          final data = generateScenariosFromScenaioOutline([
+            // pretend to be an Example section to re-use some logic
+            BddLine.fromValue(LineType.exampleTitle, ''),
+            ...dataTable,
+          ]);
+
+          for (final item in data) {
+            yield item[1];
+          }
+        }
+      } else {
+        yield lines[index];
       }
-      i += dataTable.length;
-      continue;
     }
-    yield lines[i++];
   }
-}
 
-bool _foundDataTable(List<BddLine> lines, int index) =>
-    lines[index].type == LineType.step &&
-    lines[index + 1].type == LineType.examples;
+  static bool isNextTable({
+    required List<BddLine> lines,
+    required int index,
+  }) =>
+      index < lines.length && _isTable(bddLine: lines[index]);
+
+  static bool _isTable({required BddLine bddLine}) =>
+      bddLine.type == LineType.examples;
+
+  static bool hasExamplesFormat({required BddLine bddLine}) =>
+      examplesRegExp.allMatches(bddLine.rawLine).isNotEmpty;
+
+  static List<String> _createRow({
+    required BddLine bddLine,
+  }) =>
+      List<String>.unmodifiable(
+        bddLine.value
+            .split('|')
+            .map((example) => example.trim())
+            .takeWhile((value) => value.isNotEmpty)
+            .toList(),
+      );
```

Comment:
I guess, it is safe to drop the unmodifiable usage here, and keep is simple:
```suggestion
      bddLine.value
          .split('|')
          .map((example) => example.trim())
          .takeWhile((value) => value.isNotEmpty)
          .toList();
```